#VRML_SIM R2019a utf8
WorldInfo {
}
Viewpoint {
  orientation -0.2678115869276975 0.9506847840178473 0.15644614199208362 1.0363107583680509
  position 5.363234569744613 1.7971613747378314 3.448581353806112
}
TexturedBackground {
  texture "noon_park_empty"
}
TexturedBackgroundLight {
}
DEF kedi Robot {
  translation 0 0.1 0.8100000000000003
  scale 0.75 0.75 0.75
  children [
    DEF FL Transform {
      translation -0.125 -0.049999999999999996 -0.1
      children [
        HingeJoint {
          jointParameters HingeJointParameters {
          }
          device [
            RotationalMotor {
              name "front_left"
            }
          ]
          endPoint Solid {
            rotation 1.2756979749780655e-16 9.813061345985119e-18 1 1.5707996938995743
            children [
              DEF wheel Shape {
                appearance Appearance {
                }
                geometry Cylinder {
                  height 0.049999999999999996
                  radius 0.05
                }
              }
            ]
            boundingObject USE wheel
            physics Physics {
            }
          }
        }
      ]
    }
    DEF FR Transform {
      translation 0.125 -0.049999999999999996 -0.1
      children [
        HingeJoint {
          jointParameters HingeJointParameters {
          }
          device [
            RotationalMotor {
              name "front_right"
            }
          ]
          endPoint Solid {
            rotation 1.2756979749780655e-16 9.813061345985119e-18 1 1.5707996938995743
            children [
              DEF wheel Shape {
                appearance Appearance {
                }
                geometry Cylinder {
                  height 0.049999999999999996
                  radius 0.05
                }
              }
            ]
            name "solid(1)"
            boundingObject USE wheel
            physics Physics {
            }
          }
        }
      ]
    }
    DEF RL Transform {
      translation -0.125 -0.049999999999999996 0.1
      children [
        HingeJoint {
          jointParameters HingeJointParameters {
          }
          device [
            RotationalMotor {
              name "rear_left"
            }
          ]
          endPoint Solid {
            rotation 1.2756979749780655e-16 9.813061345985119e-18 1 1.5707996938995743
            children [
              DEF wheel Shape {
                appearance Appearance {
                }
                geometry Cylinder {
                  height 0.049999999999999996
                  radius 0.05
                }
              }
            ]
            name "solid(2)"
            boundingObject USE wheel
            physics Physics {
            }
          }
        }
      ]
    }
    DEF RR Transform {
      translation 0.125 -0.049999999999999996 0.1
      children [
        HingeJoint {
          jointParameters HingeJointParameters {
          }
          device [
            RotationalMotor {
              name "rear_right"
            }
          ]
          endPoint Solid {
            rotation 1.2756979749780655e-16 9.813061345985119e-18 1 1.5707996938995743
            children [
              DEF wheel Shape {
                appearance Appearance {
                }
                geometry Cylinder {
                  height 0.049999999999999996
                  radius 0.05
                }
              }
            ]
            name "solid(3)"
            boundingObject USE wheel
            physics Physics {
            }
          }
        }
      ]
    }
    DEF BODY Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0.23215075913633937 0.6952925917448691 0.28946364538033115
        }
      }
      geometry Box {
        size 0.2 0.1 0.30000000000000004
      }
    }
  ]
  name "robot(1)"
  boundingObject USE BODY
  physics Physics {
  }
  controller "4wheel_diff_drive"
  synchronization FALSE
  cpuConsumption 100
}
DEF kedi Robot {
  translation 0.6000000000000003 0.1 0.7800000000000001
  scale 0.75 0.75 0.75
  children [
    DEF FL Transform {
      translation -0.125 -0.049999999999999996 -0.1
      children [
        HingeJoint {
          jointParameters HingeJointParameters {
          }
          device [
            RotationalMotor {
              name "front_left"
            }
          ]
          endPoint Solid {
            rotation 1.2756979749780655e-16 9.813061345985119e-18 1 1.5707996938995743
            children [
              DEF wheel Shape {
                appearance Appearance {
                }
                geometry Cylinder {
                  height 0.049999999999999996
                  radius 0.05
                }
              }
            ]
            boundingObject USE wheel
            physics Physics {
            }
          }
        }
      ]
    }
    DEF FR Transform {
      translation 0.125 -0.049999999999999996 -0.1
      children [
        HingeJoint {
          jointParameters HingeJointParameters {
          }
          device [
            RotationalMotor {
              name "front_right"
            }
          ]
          endPoint Solid {
            rotation 1.2756979749780655e-16 9.813061345985119e-18 1 1.5707996938995743
            children [
              DEF wheel Shape {
                appearance Appearance {
                }
                geometry Cylinder {
                  height 0.049999999999999996
                  radius 0.05
                }
              }
            ]
            name "solid(1)"
            boundingObject USE wheel
            physics Physics {
            }
          }
        }
      ]
    }
    DEF RL Transform {
      translation -0.125 -0.049999999999999996 0.1
      children [
        HingeJoint {
          jointParameters HingeJointParameters {
          }
          device [
            RotationalMotor {
              name "rear_left"
            }
          ]
          endPoint Solid {
            rotation 1.2756979749780655e-16 9.813061345985119e-18 1 1.5707996938995743
            children [
              DEF wheel Shape {
                appearance Appearance {
                }
                geometry Cylinder {
                  height 0.049999999999999996
                  radius 0.05
                }
              }
            ]
            name "solid(2)"
            boundingObject USE wheel
            physics Physics {
            }
          }
        }
      ]
    }
    DEF RR Transform {
      translation 0.125 -0.049999999999999996 0.1
      children [
        HingeJoint {
          jointParameters HingeJointParameters {
          }
          device [
            RotationalMotor {
              name "rear_right"
            }
          ]
          endPoint Solid {
            rotation 1.2756979749780655e-16 9.813061345985119e-18 1 1.5707996938995743
            children [
              DEF wheel Shape {
                appearance Appearance {
                }
                geometry Cylinder {
                  height 0.049999999999999996
                  radius 0.05
                }
              }
            ]
            name "solid(3)"
            boundingObject USE wheel
            physics Physics {
            }
          }
        }
      ]
    }
    DEF BODY Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0.18359655146105136 0.22548256656748303 0.7072098878461891
        }
      }
      geometry Box {
        size 0.2 0.1 0.30000000000000004
      }
    }
  ]
  name "robot(3)"
  boundingObject USE BODY
  physics Physics {
  }
  controller "4wheel_diff_drive"
  synchronization FALSE
  cpuConsumption 100
}
DEF kedi Robot {
  translation 0.7000000000000003 0.1 -0.1400000000000002
  scale 0.75 0.75 0.75
  children [
    DEF FL Transform {
      translation -0.125 -0.049999999999999996 -0.1
      children [
        HingeJoint {
          jointParameters HingeJointParameters {
          }
          device [
            RotationalMotor {
              name "front_left"
            }
          ]
          endPoint Solid {
            rotation 1.2756979749780655e-16 9.813061345985119e-18 1 1.5707996938995743
            children [
              DEF wheel Shape {
                appearance Appearance {
                }
                geometry Cylinder {
                  height 0.049999999999999996
                  radius 0.05
                }
              }
            ]
            boundingObject USE wheel
            physics Physics {
            }
          }
        }
      ]
    }
    DEF FR Transform {
      translation 0.125 -0.049999999999999996 -0.1
      children [
        HingeJoint {
          jointParameters HingeJointParameters {
          }
          device [
            RotationalMotor {
              name "front_right"
            }
          ]
          endPoint Solid {
            rotation 1.2756979749780655e-16 9.813061345985119e-18 1 1.5707996938995743
            children [
              DEF wheel Shape {
                appearance Appearance {
                }
                geometry Cylinder {
                  height 0.049999999999999996
                  radius 0.05
                }
              }
            ]
            name "solid(1)"
            boundingObject USE wheel
            physics Physics {
            }
          }
        }
      ]
    }
    DEF RL Transform {
      translation -0.125 -0.049999999999999996 0.1
      children [
        HingeJoint {
          jointParameters HingeJointParameters {
          }
          device [
            RotationalMotor {
              name "rear_left"
            }
          ]
          endPoint Solid {
            rotation 1.2756979749780655e-16 9.813061345985119e-18 1 1.5707996938995743
            children [
              DEF wheel Shape {
                appearance Appearance {
                }
                geometry Cylinder {
                  height 0.049999999999999996
                  radius 0.05
                }
              }
            ]
            name "solid(2)"
            boundingObject USE wheel
            physics Physics {
            }
          }
        }
      ]
    }
    DEF RR Transform {
      translation 0.125 -0.049999999999999996 0.1
      children [
        HingeJoint {
          jointParameters HingeJointParameters {
          }
          device [
            RotationalMotor {
              name "rear_right"
            }
          ]
          endPoint Solid {
            rotation 1.2756979749780655e-16 9.813061345985119e-18 1 1.5707996938995743
            children [
              DEF wheel Shape {
                appearance Appearance {
                }
                geometry Cylinder {
                  height 0.049999999999999996
                  radius 0.05
                }
              }
            ]
            name "solid(3)"
            boundingObject USE wheel
            physics Physics {
            }
          }
        }
      ]
    }
    DEF BODY Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0.7170366979476616 0.024582284275577935 0.09945830472266727
        }
      }
      geometry Box {
        size 0.2 0.1 0.30000000000000004
      }
    }
  ]
  name "robot(4)"
  boundingObject USE BODY
  physics Physics {
  }
  controller "4wheel_diff_drive"
  controllerArgs "1"
  synchronization FALSE
  cpuConsumption 100
}
DEF kedi Robot {
  translation 0.020000000000000025 0.1 -0.0400000000000002
  scale 0.75 0.75 0.75
  children [
    DEF FL Transform {
      translation -0.125 -0.049999999999999996 -0.1
      children [
        HingeJoint {
          jointParameters HingeJointParameters {
          }
          device [
            RotationalMotor {
              name "front_left"
            }
          ]
          endPoint Solid {
            rotation 1.2756979749780655e-16 9.813061345985119e-18 1 1.5707996938995743
            children [
              DEF wheel Shape {
                appearance Appearance {
                }
                geometry Cylinder {
                  height 0.049999999999999996
                  radius 0.05
                }
              }
            ]
            boundingObject USE wheel
            physics Physics {
            }
          }
        }
      ]
    }
    DEF FR Transform {
      translation 0.125 -0.049999999999999996 -0.1
      children [
        HingeJoint {
          jointParameters HingeJointParameters {
          }
          device [
            RotationalMotor {
              name "front_right"
            }
          ]
          endPoint Solid {
            rotation 1.2756979749780655e-16 9.813061345985119e-18 1 1.5707996938995743
            children [
              DEF wheel Shape {
                appearance Appearance {
                }
                geometry Cylinder {
                  height 0.049999999999999996
                  radius 0.05
                }
              }
            ]
            name "solid(1)"
            boundingObject USE wheel
            physics Physics {
            }
          }
        }
      ]
    }
    DEF RL Transform {
      translation -0.125 -0.049999999999999996 0.1
      children [
        HingeJoint {
          jointParameters HingeJointParameters {
          }
          device [
            RotationalMotor {
              name "rear_left"
            }
          ]
          endPoint Solid {
            rotation 1.2756979749780655e-16 9.813061345985119e-18 1 1.5707996938995743
            children [
              DEF wheel Shape {
                appearance Appearance {
                }
                geometry Cylinder {
                  height 0.049999999999999996
                  radius 0.05
                }
              }
            ]
            name "solid(2)"
            boundingObject USE wheel
            physics Physics {
            }
          }
        }
      ]
    }
    DEF RR Transform {
      translation 0.125 -0.049999999999999996 0.1
      children [
        HingeJoint {
          jointParameters HingeJointParameters {
          }
          device [
            RotationalMotor {
              name "rear_right"
            }
          ]
          endPoint Solid {
            rotation 1.2756979749780655e-16 9.813061345985119e-18 1 1.5707996938995743
            children [
              DEF wheel Shape {
                appearance Appearance {
                }
                geometry Cylinder {
                  height 0.049999999999999996
                  radius 0.05
                }
              }
            ]
            name "solid(3)"
            boundingObject USE wheel
            physics Physics {
            }
          }
        }
      ]
    }
    DEF BODY Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0.3333333333333333 0
        }
      }
      geometry Box {
        size 0.2 0.1 0.30000000000000004
      }
    }
  ]
  name "robot(5)"
  boundingObject USE BODY
  physics Physics {
  }
  controller "4wheel_diff_drive"
  synchronization FALSE
  cpuConsumption 100
}
DEF BALLS Group {
  children [
    Ball {
      translation -0.5300000000000005 0.3124999999999997 0.15
      rotation 1 0 0 4.974188353
      radius 0.036
    }
  ]
}
DEF ARENA Group {
  children [
    DEF WALLS Group {
      children [
        Wall {
          translation 1.4626794914910903 0.1 -1.2653589845047644
          rotation 0 1 0 5.7595865139999995
          name "wall(4)"
          size 1.25 0.1 0.1
          appearance Roughcast {
            colorOverride 1 0 0
            textureTransform TextureTransform {
              scale 1 2.4
            }
          }
        }
        Wall {
          translation -1.2741154269648798 0.1 1.444641017755332
          rotation 0 1 0 2.0943950960000004
          name "wall(5)"
          size 1.25 0.1 0.1
          appearance Roughcast {
            colorOverride 0 0 1
            textureTransform TextureTransform {
              scale 1 2.4
            }
          }
        }
        Wall {
          translation 0.9999999999886623 0 -1.7500067342093553
          rotation 0 1 0 1.5707996938995747
          size 0.5 0.5 0.02
        }
        Wall {
          translation -1.75 0 1
          name "wall(1)"
          size 0.5 0.5 0.02
        }
        Wall {
          translation -0.5100000000000001 0 -1.75
          name "wall(2)"
          size 3 0.5 0.5
        }
        Wall {
          translation -1.7499965992242752 0 -0.49000000000572513
          rotation 0 1 0 1.5707996938995747
          name "wall(3)"
          size 3 0.5 0.5
        }
      ]
    }
    RectangleArena {
      floorSize 4 4
      floorAppearance CarpetFibers {
        colorOverride 0.6666666666666666 0.6666666666666666 0.4980392156862745
      }
    }
  ]
}
DEF SV_BALL Robot {
  translation 0.1 0 0.1
  controller "ball_supervisor"
  supervisor TRUE
}
DEF CONVEYOR Robot {
  children [
    DEF REDTRACK Track {
      translation 1.5 0 -0.1
      rotation 0 1 0 -1.5708
      children [
        DEF BELT Shape {
          appearance Appearance {
            material Material {
              diffuseColor 1 0.39215686274509803 0.39215686274509803
            }
          }
          geometry Box {
            size 3 0.01 0.3
          }
        }
      ]
      name "track(1)"
      boundingObject USE BELT
      physics Physics {
      }
      device [
        LinearMotor {
          name "redbeltmotor"
        }
      ]
    }
    DEF BLUETRACK Track {
      translation -0.1 0 1.5
      children [
        DEF BELT Shape {
          appearance Appearance {
            material Material {
              diffuseColor 0.39215686274509803 0.39215686274509803 1
            }
          }
          geometry Box {
            size 3 0.01 0.3
          }
        }
      ]
      name "track(2)"
      boundingObject USE BELT
      physics Physics {
      }
      device [
        LinearMotor {
          name "bluebeltmotor"
        }
      ]
    }
  ]
  name "robot(2)"
  controller "belt_controller"
}
DEF CONTROLLER_SERVER Robot {
  name "robot(6)"
  controller "joystick_server"
  supervisor TRUE
  synchronization FALSE
  cpuConsumption 100
}
DEF CONTROLLER_CLIENT Robot {
  name "robot(7)"
  controller "joystick_client"
  supervisor TRUE
  synchronization FALSE
  cpuConsumption 100
}
